packages/apollo-forest-run/src/telemetry/updateStats/updateLogger.ts

```diff
@@ -0,0 +1,155 @@
+import { Draft } from "../../forest/types";
+import {
+  CompositeListChunk,
+  CompositeValueChunk,
+  ObjectChunk,
+  ValueKind,
+} from "../../values/types";
+import { CopyStats, ChunkUpdateStats, UpdateTreeStats } from "./types";
+
+function increaseObjectStats(
+  stats: CopyStats | undefined,
+  copiedFields: number,
+): void {
+  if (!stats) {
+    return;
+  }
+  stats.objectsCopied++;
+  stats.objectFieldsCopied += copiedFields;
+}
+
+function increaseArrayStats(
+  stats: CopyStats | undefined,
+  copiedItems: number,
+): void {
+  if (!stats) {
+    return;
+  }
+  stats.arraysCopied++;
+  stats.arrayItemsCopied += copiedItems;
+}
+
+export function makeCopyStats(): CopyStats {
+  return {
+    arraysCopied: 0,
+    arrayItemsCopied: 0,
+    objectFieldsCopied: 0,
+    objectsCopied: 0,
+  };
+}
+
+export class UpdateLogger {
```

Comment:
For code style consistency, I would prefer to not use classes here. But this is not urgent, so up to you for this iteration.

Reply:
I get inspirated from already existing stats collectors: https://github.com/microsoft/graphitation/blob/user/pavelglac/update-metadata/packages/apollo-forest-run/src/cache/stats.ts. I would prefer to skip the refactor in this iteration